mozilla/activity-stream
# feat(performance): #2539 Add perf telemetry + infrastructure

This adds basic performance telemetry infrastructure and the first bits of telemetry that use it for the new tab.  I suspect it'll mostly be easier to review the individual commits.

How to test if it's working

1. Edit [system-addon/lib/Activity-Stream.jsm](system-addon/lib/Activity-Stream.jsm) and 
set the `telemetry.log` preference value to `true`.
2. `npm run startmc`
3. `cd ../mozilla-central && ./mach build && ./mach run`
4. Open the Browser Console
5. Create a new tab
6. Close it
7. Make sure the logged session ping has an appropriate .perf object
